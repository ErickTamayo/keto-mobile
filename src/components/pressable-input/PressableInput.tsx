import React from 'react'
import { Text, View, GestureResponderEvent, TouchableOpacity, Image } from 'react-native'
import {
  containerStyle,
  subContainerStyle,
  iconStyle,
  textStyle,
  placeholderStyle,
  inputStyle,
  closeButtonStyle,
  closeImageStyle,
  closeDisabledImageStyle,
} from './styles'

export interface Props {
  icon: 'calendar' | 'height' | 'scale' | 'goal-scale'
  placeholder?: string
  value: string
  name: string
  onPress?: (event: GestureResponderEvent) => void
  onClear?: (event: GestureResponderEvent) => void
}

const icons = {
  calendar: require('../../assets/images/icon-calendar.png'),
  height: require('../../assets/images/icon-height.png'),
  scale: require('../../assets/images/icon-scale.png'),
  'goal-scale': require('../../assets/images/icon-goal-scale.png'),
}

const clearIcon = require('../../assets/images/icon-clear-button.png')

const PressableInput = ({ placeholder, value, onPress, onClear, icon }: Props) => (
  <View style={containerStyle}>
    <View style={subContainerStyle}>
      <View style={iconStyle}>
        <Image source={icons[icon]} />
      </View>
      <TouchableOpacity style={inputStyle} onPress={onPress}>
        <View>
          {value ? (
            <Text style={textStyle}>{value}</Text>
          ) : (
            <Text style={placeholderStyle}>{placeholder}</Text>
          )}
        </View>
      </TouchableOpacity>
      <View>
        <TouchableOpacity style={closeButtonStyle} onPress={onClear} disabled={value === ''}>
          {value ? (
            <Image style={closeImageStyle} source={clearIcon} />
          ) : (
            <View style={closeDisabledImageStyle} />
          )}
        </TouchableOpacity>
      </View>
    </View>
  </View>
)

export default PressableInput
