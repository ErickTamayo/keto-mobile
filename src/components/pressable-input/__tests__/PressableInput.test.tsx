import 'react-native'
import React from 'react'
import { TouchableOpacity, GestureResponderEvent } from 'react-native'
import { shallow } from 'enzyme'
import PressableInput, { Props } from '../PressableInput'

describe('<PressableInput />', () => {
  it('renders correctly', () => {
    const props: Props = {
      icon: 'calendar',
      placeholder: 'placeholder-text',
      value: 'value-text',
      name: 'input-name',
    }

    const wrapper = shallow(<PressableInput {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the placeholder when value is empty', () => {
    const props: Props = {
      icon: 'calendar',
      placeholder: 'placeholder-text',
      value: '',
      name: 'input-name',
    }

    const wrapper = shallow(<PressableInput {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('triggers onPress once input is pressed', () => {
    const props: Props = {
      icon: 'calendar',
      placeholder: 'placeholder-text',
      value: 'value-text',
      name: 'input-name',
      onPress: jest.fn(),
      onClear: jest.fn(),
    }

    const wrapper = shallow(<PressableInput {...props} />)
    wrapper
      .find(TouchableOpacity)
      .at(0)
      .prop('onPress')!({} as GestureResponderEvent)

    expect(props.onPress).toBeCalled()
    expect(props.onClear).not.toBeCalled()
  })

  it('triggers onClear once clear icon is pressed', () => {
    const props: Props = {
      icon: 'calendar',
      placeholder: 'placeholder-text',
      value: 'value-text',
      name: 'input-name',
      onPress: jest.fn(),
      onClear: jest.fn(),
    }

    const wrapper = shallow(<PressableInput {...props} />)
    wrapper
      .find(TouchableOpacity)
      .at(1)
      .prop('onPress')!({} as GestureResponderEvent)

    expect(props.onPress).not.toBeCalled()
    expect(props.onClear).toBeCalled()
  })
})
