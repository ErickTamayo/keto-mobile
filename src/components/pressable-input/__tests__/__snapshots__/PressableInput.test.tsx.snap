// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PressableInput /> renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "borderWidth": 1,
      },
      Object {
        "borderColor": "#57ae5b",
      },
      Object {
        "width": 240,
      },
      Object {
        "marginBottom": 13,
      },
      Object {
        "borderRadius": 5,
      },
      Object {
        "borderColor": "#e1e1e1",
      },
      Object {
        "height": 40,
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "height": 32,
          },
          Object {
            "width": 32,
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Component
        source={1}
      />
    </Component>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "width": 168,
          },
          Object {
            "height": 40,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Component>
        <Component
          style={
            Array [
              Object {
                "fontFamily": "avenir-next-medium",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#7e7e7e",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          value-text
        </Component>
      </Component>
    </TouchableOpacity>
    <Component>
      <TouchableOpacity
        activeOpacity={0.2}
        disabled={false}
        style={
          Array [
            Object {
              "height": 32,
            },
            Object {
              "width": 32,
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <Component
          source={1}
          style={
            Array [
              Object {
                "marginLeft": 13,
              },
              Object {
                "marginRight": 13,
              },
            ]
          }
        />
      </TouchableOpacity>
    </Component>
  </Component>
</Component>
`;

exports[`<PressableInput /> renders the placeholder when value is empty 1`] = `
<Component
  style={
    Array [
      Object {
        "borderWidth": 1,
      },
      Object {
        "borderColor": "#57ae5b",
      },
      Object {
        "width": 240,
      },
      Object {
        "marginBottom": 13,
      },
      Object {
        "borderRadius": 5,
      },
      Object {
        "borderColor": "#e1e1e1",
      },
      Object {
        "height": 40,
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "height": 32,
          },
          Object {
            "width": 32,
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Component
        source={1}
      />
    </Component>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "width": 168,
          },
          Object {
            "height": 40,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Component>
        <Component
          style={
            Array [
              Object {
                "fontFamily": "avenir-next-medium",
              },
              Object {
                "color": "#e1e1e1",
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "fontSize": 12,
              },
            ]
          }
        >
          placeholder-text
        </Component>
      </Component>
    </TouchableOpacity>
    <Component>
      <TouchableOpacity
        activeOpacity={0.2}
        disabled={true}
        style={
          Array [
            Object {
              "height": 32,
            },
            Object {
              "width": 32,
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <Component
          style={
            Array [
              Object {
                "marginLeft": 13,
              },
              Object {
                "marginRight": 13,
              },
            ]
          }
        />
      </TouchableOpacity>
    </Component>
  </Component>
</Component>
`;
